test/mongodb.test.js

```diff
@@ -1438,6 +1439,156 @@ describe('mongodb connector', function() {
       });
     });
 
+    it('find should be able to query by location with logical operator', function(done) {
+      var coords = { lat: 1.25, lng: 20.20 };
+      var title = 'geoWithLogicalOperatorTest';
+
+      geoDb.autoupdate(function(err) {
+        var titleIndex = 0;
+
+        var createPost = function(callback) {
+          var point = new GeoPoint({
+            lat: (Math.random() * 180) - 90,
+            lng: (Math.random() * 360) - 180,
+          });
+
+          PostWithLocation.create({
+            location: point,
+            title: title + titleIndex,
+          }, callback);
+
+          titleIndex++;
+        };
+
+        async.parallel([
+          createPost.bind(null),
+          createPost.bind(null),
+          createPost.bind(null),
+          createPost.bind(null),
+        ], function(err) {
+          should.not.exist(err);
```

Comment:
Let's refactor the error assertion as `if (err) return done(err)`, example: https://github.com/strongloop/loopback-datasource-juggler/blob/e3b6b7891c009354f36f7f811428149c61439c71/test/manipulation.test.js#L287

> should.not.exist(err)` fails the test with something like "expected Error to not exist", but callback error will enable the test runner to show the original error message and stack trace.